frontend/amundsen_application/static/js/pages/HomePage/index.tsx

```diff
@@ -29,6 +24,43 @@ export interface DispatchFromProps {
 
 export type HomePageProps = DispatchFromProps & RouteComponentProps<any>;
 
+const getHomePageWidgetComponents = (
+  layout: HomePageWidgetsConfig
+): React.ReactNode[] => {
+  /* Looks up each Widget in layout by its name property, and if found, 
+  puts the relevant component's JSX into the output array. */
+
+  const res: React.ReactNode[] = [];
+
+  layout.widgets.forEach((widget) => {
+    const WidgetComponent = React.lazy(
+      () =>
+        import('/js/features/HomePageWidgets/' + widget.options.path + '.tsx')
+    );
+
+    const additionalProps = widget.options.additionalProps
+      ? widget.options.additionalProps
+      : null;
+
+    res.push(
+      <div className="home-element-container">
+        <React.Suspense fallback={<div>Loading...</div>}>
```

Comment:
We'll probably need to create a more elaborate loading state here. How does it show now?

We should ask Knowl for something here

Reply:
Right now, I never see it actually do the fallback. Some of the components like TagsListWidget have their own more elaborate loading states (the tags list has a blur effect) which still work inside the React.Suspense.